xlators/storage/posix/src/posix-inode-fd-ops.c

```diff
@@ -2529,6 +2570,9 @@ posix_release(xlator_t *this, fd_t *fd)
     VALIDATE_OR_GOTO(this, out);
     VALIDATE_OR_GOTO(fd, out);
 
+    if (fd->inode->fd_count == 1)
+        posix_unlink_renamed_file(this, fd->inode);
```

Comment:
If the file has no open fd's and there's no entry pointing to the inode, the inode itself should be forgotten and the file unlinked in `posix_forget()`. Otherwise it means that we are holding an unnecessary reference and wasting memory (probably that's the case right now, so this is needed for now, but I would like you to add a comment here about this).

Reply:
Thinking to rename "posix_unlink_renamed_file" to something like "posix_check_free_ctx" as prior also does the freeing of the context